modal/_asgi.py

```diff
@@ -204,45 +204,54 @@ async def request_generator():
                 if not message.get("more_body", False):
                     break
             elif message["type"] == "http.disconnect":
-                proxy_response.connection.transport.abort()  # Abort the connection.
-                break
+                if proxy_response:
+                    proxy_response.connection.transport.abort()  # Abort the connection.
+                else:
+                    raise NameError("Proxy response not defined")
             else:
                 raise ExecutionError(f"Unexpected message type: {message['type']}")
 
     path = scope["path"]
     if scope.get("query_string"):
         path += "?" + scope["query_string"].decode()
 
-    proxy_response = await session.request(
-        method=scope["method"],
-        url=path,
-        headers=[(k.decode(), v.decode()) for k, v in scope["headers"]],
-        data=None if scope["method"] in aiohttp.ClientRequest.GET_METHODS else request_generator(),
-        allow_redirects=False,
-    )
-
-    async def send_response():
-        msg = {
-            "type": "http.response.start",
-            "status": proxy_response.status,
-            "headers": [(k.encode(), v.encode()) for k, v in proxy_response.headers.items()],
-        }
-        await send(msg)
-        async for data in proxy_response.content.iter_any():
-            msg = {"type": "http.response.body", "body": data, "more_body": True}
+    try:
+        proxy_response = await session.request(
+            method=scope["method"],
+            url=path,
+            headers=[(k.decode(), v.decode()) for k, v in scope["headers"]],
+            data=None if scope["method"] in aiohttp.ClientRequest.GET_METHODS else request_generator(),
+            allow_redirects=False,
+        )
+    except aiohttp.ClientConnectionError as e:
+        if isinstance(e.__cause__, NameError):
+            return
+        else:
+            raise e.__cause__
+    else:
```

Comment:
I don't think you need the `else:` block here because the `except` code path will raise or return. Generally it could be better to avoid indenting a large block of continuation code since that makes following it nonlinear.